app/controllers/vm_common.rb

```diff
@@ -1186,11 +1214,28 @@ def open_parent_nodes(record)
     add_nodes
   end
 
+  # if node is VM or Template and hide_vms is true - select parent node in explorer tree but show info of Vm/Template
+  def resolve_node_info(id)
+    nodetype, id = id.split("-")
+
+    if hide_vms && (nodetype == 'v' || nodetype == 't')
+      self.x_node = parent_folder_id(VmOrTemplate.find(id))
+      @vm = VmOrTemplate.find(id)
```

Comment:
Also feel like the ui very often converts a class into an id an then calls find to get the object back.
probably not the case here, but it feels like a pattern